File: app/src/main/java/com/technolifestyle/therealshabi/brogrammar/SharedPreferenceUtils/SharedPreferenceStorage.java
package com.technolifestyle.therealshabi.brogrammar.SharedPreferenceUtils;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by shahbaz on 25/2/17.
 */

public class SharedPreferenceStorage {
    public static final String MY_SHARED_PREFERENCES = "my_shared_preferences";
    public static final String MAIN_ACTIVITY_FLAG = "main_activity_flag";
    public static final String ACCESS_CODE = "access_code";
    public static final String USER_ID = "user_id";
    public static final String TEAM_NAME = "team_name";

    public static void setSharedPreferenceMainActivityFlag(Context context, boolean flag) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(MY_SHARED_PREFERENCES, context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putBoolean(MAIN_ACTIVITY_FLAG, flag);
        editor.commit();
    }

    public static boolean getSharedPreferenceMainActivity(Context context) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(MY_SHARED_PREFERENCES, context.MODE_PRIVATE);
        return sharedPreferences.getBoolean(MAIN_ACTIVITY_FLAG, false);
    }

    public static void setAccessCode(Context context, String access_code) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(MY_SHARED_PREFERENCES, context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.clear();
        editor.putString(ACCESS_CODE, access_code);
        editor.commit();
    }

    public static String getAccessCode(Context context) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(MY_SHARED_PREFERENCES, context.MODE_PRIVATE);
        return sharedPreferences.getString(ACCESS_CODE, null);
    }

    public static void setUserId(Context context, String user_id) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(MY_SHARED_PREFERENCES, context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(USER_ID, user_id);
        editor.commit();
    }

    public static String getUserId(Context context) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(MY_SHARED_PREFERENCES, context.MODE_PRIVATE);
        return sharedPreferences.getString(USER_ID, null);
    }

    public static void setTeamName(Context context, String teamName) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(MY_SHARED_PREFERENCES, context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(TEAM_NAME, teamName);
        editor.commit();
    }

    public static String getTeamName(Context context) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(MY_SHARED_PREFERENCES, context.MODE_PRIVATE);
        return sharedPreferences.getString(TEAM_NAME, null);
    }
}

File: app/src/main/java/com/technolifestyle/therealshabi/brogrammar/sharedPreferenceUtils/SharedPreferenceStorage.kt
package com.technolifestyle.therealshabi.brogrammar.sharedPreferenceUtils

import android.content.Context
import android.content.Context.MODE_PRIVATE

/**
 * Created by shahbaz on 25/2/17.
 */

object SharedPreferenceStorage {
    private const val MY_SHARED_PREFERENCES = "my_shared_preferences"
    private const val MAIN_ACTIVITY_FLAG = "main_activity_flag"
    private const val ACCESS_CODE = "access_code"
    private const val USER_ID = "user_id"
    private const val TEAM_NAME = "team_name"

    fun setSharedPreferenceMainActivityFlag(context: Context, flag: Boolean) {
        val sharedPreferences = context.getSharedPreferences(
                MY_SHARED_PREFERENCES, MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        editor.putBoolean(MAIN_ACTIVITY_FLAG, flag)
        editor.apply()
    }

    fun getSharedPreferenceMainActivity(context: Context): Boolean {
        val sharedPreferences = context.getSharedPreferences(
                MY_SHARED_PREFERENCES, MODE_PRIVATE)
        return sharedPreferences.getBoolean(MAIN_ACTIVITY_FLAG, false)
    }

    fun setAccessCode(context: Context, access_code: String) {
        val sharedPreferences = context.getSharedPreferences(
                MY_SHARED_PREFERENCES, MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        editor.putString(ACCESS_CODE, "")
        editor.putString(ACCESS_CODE, access_code)
        editor.apply()
    }

    fun getAccessCode(context: Context): String? {
        val sharedPreferences = context.getSharedPreferences(
                MY_SHARED_PREFERENCES, MODE_PRIVATE)
        return sharedPreferences.getString(ACCESS_CODE, null)
    }

    fun setUserId(context: Context, user_id: String) {
        val sharedPreferences = context.getSharedPreferences(
                MY_SHARED_PREFERENCES, MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        editor.putString(USER_ID, user_id)
        editor.apply()
    }

    fun getUserId(context: Context): String? {
        val sharedPreferences = context.getSharedPreferences(
                MY_SHARED_PREFERENCES, MODE_PRIVATE)
        return sharedPreferences.getString(USER_ID, null)
    }

    fun setTeamName(context: Context, teamName: String) {
        val sharedPreferences = context.getSharedPreferences(
                MY_SHARED_PREFERENCES, MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        editor.putString(TEAM_NAME, teamName)
        editor.apply()
    }

    fun getTeamName(context: Context): String? {
        val sharedPreferences = context.getSharedPreferences(
                MY_SHARED_PREFERENCES, MODE_PRIVATE)
        return sharedPreferences.getString(TEAM_NAME, null)
    }
}
